pds_doi_core/actions/draft.py

```diff
@@ -83,30 +148,39 @@ def run(self,
             logger.info(f"permissible_contributor_list {o_permissible_contributor_list}")
             exit(1)
 
-        # parse input
-        if not input.startswith('http'):
-            xml_tree = etree.parse(input)
-        else:
-            response = requests.get(input)
-            xml_tree = etree.fromstring(response.content)
+        # The value of input can be a list of names, or a directory.  Resolve that to a list of names.
+        list_of_names = self._resolve_input_into_list_of_names(input)
+        # Return immediately if there are no files found.
+        # Note that exit() function is not used as it would interfere with the unit test.
+        if len(list_of_names) == 0:
+            logger.error(f"Provided input parameter results in no file names parsed or input directory {input} is empty.")
+            # Return empty tree.
+            o_doi_label = etree.Element("records")
+            return etree.tostring(o_doi_label,pretty_print=True).decode()
 
-        doi_fields = self.m_doi_pds4_label.get_doi_fields_from_pds4(xml_tree)
-        doi_fields['publisher'] = self._config.get('OTHER', 'doi_publisher')
-        doi_fields['contributor'] = contributor_value
+        # Create an empty tree with 'records' as the root tag.
+        # An element will be added from the output of each file parsed.
+        o_doi_labels = etree.Element("records") # OSTI uses 'records' as the root tag.
 
+        # Batch processing logic:
+        # For each name found, transform the PDS4 label to an OSTI record, then concatenate that record to o_doi_label to return. 
 
-        # generate output
-        o_doi_label = self.m_doi_output_osti.create_osti_doi_draft_record(doi_fields)
+        for input_file in list_of_names:
 
-        # Use the service of TransactionBuilder to prepare all things related to writing a transaction.
-        transaction_obj = self.m_transaction_builder.prepare_transaction(input,
-                                                                         node,
-                                                                         submitter,
-                                                                         [doi_fields],
-                                                                         output_content=o_doi_label)
+            # Transform the PDS4 label to an OSTI record.
+            doi_label = self._transform_pds4_label_into_osti_record(input_file,node,submitter,contributor_value)
 
-        # Write a transaction for the 'reserve' action.
-        transaction_obj.log()
+            # The returned label can be None if it is not an expected XML file.
+            if doi_label:
```

Comment:
This test looks like an implicit error management. That would be more readable to have an explicit exception (invalid PDS4Input...) raised by the method _transform_pds4_label_into_osti_record().
But 
I don't know what case would raise this exception. Do you know what case can make the doi_abel == None ?

Reply:
To avoid having to check for doi_label be None, the function _transform_pds4_label_into_osti_record() can return an empty tree (which is not None).  There's no need to check for None-ness.

The code is much simpler now.